### PR TITLE
Block unsafe and have clippy warn on unsafe|panic

### PR DESCRIPTION
**Pull request by lukehinds**
I added exceptions to instances of unwrap that have a comment
justifying use.

We now need to either toggle the last to handling the Option / Result or
use except.

Test uses of unwrap will be ignored (as will examples)

Blocks unsafe

Signed-off-by: Luke Hinds <lhinds@redhat.com>

### CODE REVIEW

**Comment by lukehinds** [comment]
@tarcieri

File: src/lib.rs [hunk 1]
```diff
@@ -160,6 +160,8 @@
 //! Additional examples can be found inside of the [`examples`](https://github.com/sigstore/sigstore-rs/tree/main/examples/)
 //! directory.
 
+#![forbid(unsafe_code)]
```
Comment (tarcieri) [inline src/lib.rs:163]: I'd suggest sticking the clippy lint configuration here as well:

```suggestion
#![forbid(unsafe_code)]
#![warn(clippy::unwrap_used, clippy::panic)]
```

**Review by viccuad — COMMENTED** [review]
Indeed, we should do this, thanks Luke.

Do you see tackling the rest of cases yourself, or would you prefer for someone else to solve it in main before merging this?

**Comment by lukehinds** [comment]
> Indeed, we should do this, thanks Luke.
> 
> Do you see tackling the rest of cases yourself, or would you prefer for someone else to solve it in main before merging this?

It might better to get someone who knows that part of the code well @flavio I would guess.

**Comment by flavio** [comment]
@lukehinds sorry for being late to the party :cocktail:

I've pushed refactored the code and pushed the fix to your PR

**Comment by flavio** [comment]
@lukehinds your first commit on this PR is not signed, this is bothering the DCO check

**Comment by lukehinds** [comment]
thanks @flavio, fixed DCO bot now!

**Comment by flavio** [comment]
@lukehinds unfortunately the DCO is still off, you have to add the "Signed-off-by" to 322f2efb7d5eec62529e1b9585eea8d078eb23ba

**Comment by lukehinds** [comment]
ahh, sorry, done now @flavio

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -62,4 +62,4 @@ jobs:
       - uses: actions-rs/cargo@844f36862e911db73fe0815f00a4a2602c279505 # v1.0.3
         with:
           command: clippy
-          args: --workspace -- -D warnings
+          args: --workspace -- -D warnings -D clippy::unwrap_used -D clippy::panic
```
[the remark this answers is not in the document]
Comment (flavio) [reply .github/workflows/tests.yml]: Thanks @tarcieri, I've fixed that

**Pull request merged**